TodoApp/Tests/TodoApp.Services.Tests/Items/ItemUpdatingServiceTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Threading.Tasks;
+using NSubstitute;
+using NUnit.Framework;
+using TodoApp.Contracts.Repositories;
+using TodoApp.Contracts.Services;
+using TodoApp.Contracts.Wrappers;
+using TodoApp.Services.Items;
+using TodoApp.TestContracts.Extensions;
+using TodoApp.TestContracts.Factories;
+
+namespace TodoApp.Services.Tests.Items
+{
+    [TestFixture]
+    public class ItemUpdatingServiceTests
+    {
+        private IDateTimeWrapper _dateTimeWrapper;
+        private IItemRepository _itemRepository;
+        private IItemObtainingService _itemObtainingService;
+        private IItemUpdatingService _itemUpdatingService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _dateTimeWrapper = Substitute.For<IDateTimeWrapper>();
+            _itemObtainingService = Substitute.For<IItemObtainingService>();
+            _itemRepository = Substitute.For<IItemRepository>();
+
+            _itemUpdatingService = new ItemUpdatingService(_itemObtainingService, _itemRepository, _dateTimeWrapper);
+        }
+
+        [Test]
+        public async Task UpdateAsync_ExistingItem_UpdatesItem()
+        {
+            var creationTime = new DateTime(2018, 9, 27, 12, 33, 41);
+            var currentTime = new DateTime(2018, 10, 1, 8, 42, 5);
+            var id = Guid.Parse("F7148339-E162-4657-B886-C29BF6A2D312");
+            var updatedItem = ItemFactory.CreateItem(id, "This is a text.", creationTime, creationTime);
+            var changedItem = ItemFactory.CreateItem(id, "This is a text.", creationTime, currentTime);
+
+            _itemObtainingService.ExistsAsync(id).Returns(true);
+            _itemObtainingService.GetById(id).Returns(updatedItem);
+            _dateTimeWrapper.CurrentDateTime().Returns(currentTime);
+
+            await _itemUpdatingService.UpdateAsync(updatedItem);
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(updatedItem, Is.EqualTo(changedItem).UsingItemComparer());
```

Comment:
To make this test straight:
1. You need to introduce some "`receivedItem`" that represent incomplete item passed to the service from the controller (it should not have dates and times, otherwise you would make an impression that these can be altered too)
1. The `UpdateAsync` should receive "`receivedItem`"
1. The assertion relying on the `updatedItem` alteration should be omitted
1. The `Received` assertion should be performed against the `changedItem` variable
1. The text of the original and changed item should differ (`receivedItem.Text` == `changedItem.Text` && `updatedItem.Text` != `receivedItem.Text`)
1. the `updatedItem` is ambiguous, please use something that cannot refer to both initial and amended item (`originalItem`, `storedItem`, `existingItem`, `previousItem`, `initialItem`, …)